python/ray/_raylet.pyx

```diff
@@ -242,15 +248,27 @@ cdef Language LANG_CPP = Language.from_native(LANGUAGE_CPP)
 cdef Language LANG_JAVA = Language.from_native(LANGUAGE_JAVA)
 
 
-cdef unordered_map[c_string, double] resource_map_from_dict(resource_map):
+cdef int prepare_resources(
+        dict resource_dict,
+        unordered_map[c_string, double] *resource_map) except -1:
     cdef:
         unordered_map[c_string, double] out
         c_string resource_name
-    if not isinstance(resource_map, dict):
-        raise TypeError("resource_map must be a dictionary")
-    for key, value in resource_map.items():
-        out[key.encode("ascii")] = float(value)
-    return out
+
+    if resource_dict is None:
```

Comment:
I don't think this can be hit since `resource_dict` is already a dict as per the type declaration above.

Reply:
Verified that this check is hit when `None` is passed in.